### PR TITLE
Speed up Liquibase startup

### LINKED ISSUE #8942: Liquibase performance improvment
**Description**
As for now, liquibase is significantly impacting startup and dev mode performance

Based on: https://github.com/danielpetisme/sample-jhipster-quarkus-liquibase
Starup time(dev profile): ~6-7 seconds
Dev mode reload: 3 seconds

My concerns is that liquibase is compromising Quarkus speed promise. Considering a dev using Quarkus for the first time,  he/she won't understand why the difference between Quarkus and other framework startup is not as high as explained in the communication 

**Implementation ideas**
@stuartwdouglas already started some work.

### PR DESCRIPTION
**Pull request by stuartwdouglas**
Resolves: #8942

### CODE REVIEW

**Comment by stuartwdouglas** [comment]
This improves the JHipster Quarkus blueprint speedup time considerably (over 1s on my machine).

File: extensions/liquibase/deployment/pom.xml [hunk 1]
```diff
@@ -64,6 +64,72 @@
                     </annotationProcessorPaths>
                 </configuration>
             </plugin>
+
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-dependency-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id>extract-liqibase-for-indexing</id>
```
Comment (gsmet) [inline extensions/liquibase/deployment/pom.xml:73]:
```suggestion
                        <id>extract-liquibase-for-indexing</id>
```

File: extensions/liquibase/deployment/pom.xml [hunk 2]
```diff
@@ -64,6 +64,72 @@
...
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+
+            <plugin>
+                <artifactId>maven-resources-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id>copy-liqibase-index</id>
```
Comment (gsmet) [inline extensions/liquibase/deployment/pom.xml:113]:
```suggestion
                        <id>copy-liquibase-index</id>
```

File: extensions/liquibase/deployment/src/main/java/io/quarkus/liquibase/LiquibaseProcessor.java [hunk 3]
```diff
@@ -55,11 +74,92 @@
...
+                            !Modifier.isPublic(found.flags())) {
+                        continue;
+                    }
+                    AnnotationInstance annotationInstance = found.classAnnotation(liquibaseServiceName);
+                    if (annotationInstance == null || !annotationInstance.value("skip").asBoolean()) {
+                        impls.add(found.name().toString());
+                    }
+                }
+            }
+            //if we known what DB types are in use we limit them
```
Comment (gsmet) [inline extensions/liquibase/deployment/src/main/java/io/quarkus/liquibase/LiquibaseProcessor.java:143]:
```suggestion
            //if we know what DB types are in use we limit them
```

**Review by gsmet — CHANGES_REQUESTED** [review]
This looks very good except for a few typos.

File: extensions/liquibase/runtime/src/main/java/io/quarkus/liquibase/runtime/LiquibaseRecorder.java [hunk 4]
```diff
@@ -100,4 +107,28 @@ private void validate(BeanContainer container, AnnotationLiteral<? extends Annot
...
+                    log.warnf("Failed to find pre-indexed service %s, falling back to slow classpath scanning",
+                            requiredInterface);
+                    return super.findClasses(requiredInterface);
+                }
+                List<Class<? extends T>> ret = new ArrayList<>();
+                for (String i : found) {
+                    try {
+                        ret.add((Class<? extends T>) Class.forName(i, false, Thread.currentThread().getContextClassLoader()));
+                    } catch (ClassNotFoundException e) {
+                        log.error("Failed to load liqibase service", e);
```
Comment (fbricon) [inline extensions/liquibase/runtime/src/main/java/io/quarkus/liquibase/runtime/LiquibaseRecorder.java:127]: liquibase

Comment (gsmet) [inline extensions/liquibase/runtime/src/main/java/io/quarkus/liquibase/runtime/LiquibaseRecorder.java:127]:
```suggestion
                        log.error("Failed to load Liquibase service", e);
```

**Comment by gsmet** [comment]
Force pushed an update with the typo fixes.

**Review by gsmet — APPROVED** [review]

**Pull request merged**